### ISSUE TITLE
Can't save maps to sd card anymore

### ISSUE THREAD

**Issue by bbinet** [organicmaps/organicmaps#2684]
I used to save my maps to sd card, but it seems this is not possible anymore.
It says "read-only" but I can successfully write to the sdcard from other Android apps.

![Screenshot_20220606-180204_Trebuchet](https://user-images.githubusercontent.com/76281/172199941-41aba072-8fc9-4899-9409-21f3e668b4a2.png)

Note that I'm running Android 11 (Lineage OS) on Samsung Galaxy S5.

**Comment by biodranik** [comment]
Did it work for you earlier on Android 10? Or did it work on Android 11?

**Comment by bbinet** [comment]
It worked on Android 11.

**Comment by bbinet** [comment]
(I still have organicmaps files in the SD card organicmaps data location)

**Comment by biodranik** [comment]
So the only difference is that the older OM version worked, and the newer one stopped, right? Did you do any system updates?
@pastk can you please investigate this case? Maybe there is an issue in our write-check code.

**Comment by bbinet** [comment]
I don't know in which OM update this issue appeared in the beginning as FDroid updates are automatic and I rarely use OM, but I remember a few weeks ago that all my downloaded maps had disappeared (need to be downloaded again). I didn't pay attention at that time that it was a SD card issue, so the new downloads of maps went to the internal storage.

And yes, I probably did few LineageOS system updates via OTA in the meantime (but other apps like AntennaPod or the camera, which are also configured to use SD card storage still work fine).

**Comment by pastk** [comment]
Hi @bbinet could you please send us logs to help debug an issue?
Tick "Enable logging" in OM settings, then go into "Save maps to" setting, then go to "Help" and tap "Report a bug" (you can send the logs via email, or save them as a file (if your file manager offers this option) and attach here - but please be aware the logs can contain private info like your location, or if you're in OM telegram group then send them directly to me @pastk ).

**Comment by aniston** [comment]
Hello @pastk , since you are on this issue and I'm having the same problem as @bbinet on a ::
- Samsung Galaxy XCover 4S
- Android 11 with Android Security Patch May 1 2022
- Organic Maps v2022.05.31-10-FDroid with OSM vMay 15 2022

The issue is the problem not being able to save the Downloaded maps to the SD card as it is shown as "_read only_" and hence not selectable.
My App storage settings (Android System) under Settings > Apps > Organic Maps > Storage > the option Storage used is already set to "_External Storage_" (done that by "**Change**" from Internal Storage to SD
And in the App under Settings > Save maps to > no possibility of selecting anything else other than the pre selected option of "_Internal Storage_" exactly as the opening image posted at the beginning.

as requested sending the logs.zip file for analysis, .HTH
[logs.zip](https://github.com/organicmaps/organicmaps/files/8878091/logs.zip)
Looking at the storage.log file , any reason that the path is set to "Currently configured storage: /storage/emulated/0/Android/data/app.organicmaps/files/" instead of "OSM" as it was in the past ?

Although the Maps are working as they are present on the Internal storage, would like to store them to the SD as space constraints to Internal storage is high.

**Comment by pastk** [comment]
Hi @aniston thanks for the report and the logs!
So it looks like you use SD card as an adopted/extended storage (as you've moved OM to SD in sys settings)? @bbinet do you use your SD card as adopted storage as well?

1. Does the issue persist if you move OM to internal storage in system settings?

2. Do you mind installing and getting logs from [this beta build](https://drive.google.com/file/d/1Ch5qe41QftFM9EkeAd5MGeZHIQfUqOen/view?usp=sharing)?
(I'e added more verbose logging and changed the way SD writability is checked)

Thanks!

**Comment by aniston** [comment]
Hi @pastk , I'm back with my notes and [Logfile _logsBeta.zip_](https://github.com/organicmaps/organicmaps/files/8878902/logsBeta.zip) report from OM [beta](https://drive.google.com/file/d/1Ch5qe41QftFM9EkeAd5MGeZHIQfUqOen/view?usp=sharing) build ::
- OM beta installed in parallel with OM (ie: did not uninstall the current OM)
- without touching System App Settings OM Beta installs to _Internal storage_
- ![image](https://user-images.githubusercontent.com/3067891/173071512-9e0bc146-c38c-4b42-b045-f5b20e05762e.png)
- Internal storage is selected as default and downloaded maps are ending in to _\Internal storage\Android\data\app.organicmaps.beta\files\220515_
- still cannot select the SD card as Download folder although a change is noticed that the 3.rd. option is now available , though always on _Internal storage_
- ![image](https://user-images.githubusercontent.com/3067891/173071386-e06e8116-6c89-4be2-87f0-c75ef718f233.png)

so still cannot change to SD card for Downloaded Maps.

**Comment by pastk** [comment]
@aniston thanks!

As far as I can see from the release logs you attached in https://github.com/organicmaps/organicmaps/issues/2684#issuecomment-1152243568 you had 3 storage options visible already?

Anyway,
here is a new beta.
[OrganicMaps-22060901-fdroid-beta_2.apk](https://drive.google.com/file/d/1mmbx-sU1WtcCM_KuvAFKOKHoGO8UyyoK/view?usp=drive_web)
Please uninstall the previous one first.
I assume your SD card will show up as writable now. In this case please try to switch to it via the "Save maps to" and then try to download some smaller map for a test.
And send in new logs please.

**Comment by aniston** [comment]
@pastk thanks for the Beta_2, I decided to start clean and removed both OM and OM_beta_1, put in a fresh new SD card at which point Android asked me to format it without other choices (except cancel).

So with the new OM Beta_2 installed at startup it asked me to download the base Worldmap and it completes its download but is not able to save it (no error message) just that the file is not there and background world map is blank.

This is verified as every time OM beta_2 starts up i'm asked to download the base Worldmao and the _Download maps_ menu list is empty.

Under _Settings_ the _Save maps to_ now has the option _SD card_ , and is selected (option to select _Internal storage_ is also available) but apparently Maps are getting downloaded but not saved anywhere in the SD card nowhere for that matter with this option selected.

Reading around on the internet i came across some other older Apps that are facing the [similar problem of saving to SD card under Android 11](https://forums.redflagdeals.com/android-11-microsd-card-storage-new-restrictions-break-some-apps-2481628/), could this be Android 11 specific ?
- don't know if it will help , but a website point the Android Developer page specific to the theme [App-specific directory on external storage](https://developer.android.com/about/versions/11/privacy/storage)

as usual hier ist the log file [logsBeta2.zip](https://github.com/organicmaps/organicmaps/files/8879957/logsBeta2.zip)

@pastk , Thanks again for your time.

**Comment by pastk** [comment]
@aniston and @bbinet your cases look consistent so far, so might be specific to Samsung devices (or last Samsung updates?).

@aniston I'll have a look at the links you posted shortly, meanwhile could you test one more beta please? 
[ OrganicMaps-22060901-fdroid-beta-3.apk](https://drive.google.com/file/d/1DBliv93Hxp80N9Gw8gUnRZvZO0QxRClF/view?usp=drive_web)

**Comment by pastk** [comment]
@aniston ah, and if you have access to `Android/data/app.organicmaps.beta/files/` on your SD card could you please check if there are any files or subfolders there since beta2 test?

**Comment by pastk** [comment]
> > @aniston ah, and if you have access to `Android/data/app.organicmaps.beta/files/` on your SD card could you please check if there are any files or subfolders there since beta2 test?
> 
> @pastk on the SD card in _/Android/data_ there are no files at all, so I guess we cannot write there.
> 
>     * I can confirm that with the Android Phone  connected via USB to a PC I cannot write into the SD card _/Android/data_ path on the contrary i can create/copy folders & files in the root path of the SD card , seems very much like a SD card path restriction in Android 11 maybe ?

This is as expected. Since A11 apps have read-write to their app-specific directories only, so OM is expected to have access to `Android/data/app.organicmaps.beta/files/` either on SD (which it doesn't for some reason) or on internal storage (which it does as e.g. log files are saved there by default).
And its expected that other apps (and hence a user) don't have access to this directory.
(you might be able to have a read-only access to it via some file managers that hold a special "all files premission", e.g. try Total Commander)

> 
>     * also in the Internal storage of the phone in _/Android/data_ I **can** create/copy files and folders.

And this is not expected actually :)

> That said I would not mind to save the files even in the root folder of the SD card (or atleast a way to specify a folder there). If that could work as a carry-on method it should look good.

Nope, the app can't have access to root folders (or any other non-app-specific folders) anymore..

In the link you posted one user had success using SD card for maps storage with OSMAnd - do you mind to try to see if it works? Maybe they've found a workaround for this issue.

**Comment by pastk** [comment]
> 
>     * without touching System App Settings OM Beta installs to _Internal storage_
 ![image](https://user-images.githubusercontent.com/3067891/173071512-9e0bc146-c38c-4b42-b045-f5b20e05762e.png)


Can you please check what does the system info show for OM storage size?
Uninstall previous betas please and check storage sizes
1. after a fresh install but before launching an app
2. after downloading a world map
3. after downloading any other (small, just for a test) map

Thanks!

**Comment by pastk** [comment]
@aniston 

> @pastk thanks for the Beta_2, I decided to start clean and removed both OM and OM_beta_1, put in a fresh new SD card at which point Android asked me to format it without other choices (except cancel).
> 

As you have deleted a release version of OM anyway, could you please test [a March release](https://github.com/organicmaps/organicmaps/releases/download/2022.03.23-4-android/OrganicMaps-22032304-web-release.apk) too? This is to ensure its not a regression with recent changes in OM's storage code.
Thanks in advance!

**Comment by aniston** [comment]
@pastk , sorry I could not get back earlier !

1. test with OSMAnd works ie: saved the downloaded maps, only if the selected path begins with _/storage/emulated/0_ like here 
![image](https://user-images.githubusercontent.com/3067891/173156754-f4fb59b7-6481-4a18-9550-e6ead30fc23e.png)
- in the case of the chosen path starting with _/storage/54AE-180C_ then no maps get saved 
![image](https://user-images.githubusercontent.com/3067891/173157119-2561bbef-2af3-4653-8621-bb88b8b36414.png)
- same case for the Multiuser storage, so long the path starts with _/storage/emulated/0_ files can be saved.
- From the icons I'm guessing that the /emulated/ structure is not actually the External SD albeit they call it External Storage 1 !
- in contrast the External storage 2 icon is right and the path too is directly the SD card and not storing the map data.

2. After a fresh Install (with no leftovers) of the last OM Beta3 and without opening the App Storage reports 71.24 MB
- after first start and  worldmap download its 125MB
- after downloading Berlin map storage is now 215MB

3. the older version [March release](https://github.com/organicmaps/organicmaps/releases/download/2022.03.23-4-android/OrganicMaps-22032304-web-release.apk) does not have the _Save maps to_ option, so there was no choice !
![image](https://user-images.githubusercontent.com/3067891/173159157-6ea9db14-5edc-435c-949e-ce19867941d6.png)
-----------------------------------------
have to retire for the night , will catch up in 10 hours time .

**Comment by pastk** [comment]
>     * From the icons I'm guessing that the /emulated/ structure is not actually the External SD albeit they call it External Storage 1 !

you're right, "emulated" is backed by internal storage actually, its peculiarities of Android's storage naming...

>     2. After a fresh Install (with no leftovers) of the last OM Beta3 and without opening the App Storage reports 71.24 MB
> 
>     * after first start and  worldmap download its 125MB
> 
>     * after downloading Berlin map storage is now 215MB
> 

Ok this means that maps are saved to the card actually, but could not be read / not visible to the app.. Very strange :)


> 
>     3. the older version [March release](https://github.com/organicmaps/organicmaps/releases/download/2022.03.23-4-android/OrganicMaps-22032304-web-release.apk) does not have the _Save maps to_ option, so there was no choice !

Its about how I expected it to be, but anyway, just to confirm, could you please enable the logs, restart the app and then send the logs?

Thanks a lot for your help in debugging this!

**Comment by aniston** [comment]
> > ```
> > 2. After a fresh Install (with no leftovers) of the last OM Beta3 and without opening the App Storage reports 71.24 MB
> > 
> > * after first start and  worldmap download its 125MB
> > 
> > * after downloading Berlin map storage is now 215MB
> > ```
> 
> 
> Ok this means that maps are saved to the card actually, but could not be read / not visible to the app.. Very strange :)

@pastk that test is with Internal storage!  not the SD card.

**Comment by pastk** [comment]
> 
> @pastk that test is with Internal storage! not the SD card.

Are you sure? According to you beta3 logs OM chooses SD card as maps storage by default (please check in "Save maps to" settings).
(it doesn't matter what Android settings tell you, its important what is chosen in OM settings)

**Comment by aniston** [comment]
> Are you sure? According to you beta3 logs OM chooses SD card as maps storage by default (please check in "Save maps to" settings).
> (it doesn't matter what Android settings tell you, its important what is chosen in OM settings)

Hi @pastk , started again clean with Beta_3 to check.
1. Android storage says its on Internal storage.
- ![image](https://user-images.githubusercontent.com/3067891/173182410-cf5ef313-c540-4af3-bd81-54d3ae916e70.png)
----
2. App says Maps are to be stored in _/storage/54AE-180C/Android/data/app.organicmaps.beta/files_ although in the storage log we are seeing canWrite == false `2022-06-11 11:22:50.720 main: W/com.mapswithme.maps.settings.StoragePathManager:   canWrite == false: /storage/54AE-180C/Android/data/app.organicmaps.beta/files/ - external, 31858393088 available of 31906070528 bytes, removable, state=mounted, uuid=54AE-180C, label='SD card'`
- ![image](https://user-images.githubusercontent.com/3067891/173182216-9fd8fb4c-fabe-4468-926a-885b8c9de81d.png)
----
3. Every time the App is opened it asks to download the initial Worldmap  as it finds no Maps in that folder.
- ![image](https://user-images.githubusercontent.com/3067891/173182616-25aed1fe-bd02-4bc9-8ad7-15acc35dd91f.png)
- good worldmap downloaded, but still not showing under Downloaded maps
- ![image](https://user-images.githubusercontent.com/3067891/173182653-47053ff0-68bf-47e1-8836-48f9f119443f.png)
- To test I downloaded a Berlin map and it showed in the Downloaded Maps (maybe this is in cache)
- ![image](https://user-images.githubusercontent.com/3067891/173182683-ddb76ac6-5b52-4702-99c3-1b4b0531c665.png)
- ![image](https://user-images.githubusercontent.com/3067891/173182710-1a6c784a-1da1-4e08-a13a-0fa0846a7156.png)
- then exit the App and open the App again, same result no initial worldmap so have to download again
- ![image](https://user-images.githubusercontent.com/3067891/173182769-7378f1e9-b4c0-4c3a-96c7-85fd10bbc046.png)
- and this time no Berlin Map that was previously saved so Downloaded Maps list is empty although again the worldmap was downloaded.
- ![image](https://user-images.githubusercontent.com/3067891/173182867-476a0716-44a1-424f-ae68-bdd0235ec6e1.png)
----
Attaching the Beta_3 [logsBeta3_20220611.zip](https://github.com/organicmaps/organicmaps/files/8883616/logsBeta3_20220611.zip) as this time App was closed and opened atleast 3 times, maybe you can find something more in the logs.

HTH.

**Comment by pastk** [comment]
@aniston 
thanks a lot for the details!

Here is one more :)
[beta4](https://drive.google.com/file/d/1RwIl38QZnb9cpYjWy4g5m7qI7Sg74pLi/view?usp=sharing)
Please test as usual and restart the app before sending logs.

Also this time could you do 2 tests please?
1st with OM installed into Internal (default).
2nd with OM moved to External in android app settings.
In both cases please ensure SD card is selected in OM settings.

**Comment by aniston** [comment]
@pastk results from beta4 testing (ensured SD card is selected in OM settings)
1. OM Internal in Android App settings -> Maps get downloaded but on App restart not available and hence forced to download again.
2. OM moved to External in android app settings -> same situation Maps asked to download but not available after App restart.

attached Log files from beta4 [logsBeta4.zip](https://github.com/organicmaps/organicmaps/files/8883763/logsBeta4.zip)

**Comment by pastk** [comment]
@aniston @bbinet could you please tell detailed version of Android you're running?
It should be somewhere in android settings "About" or "System -> About". For a Samsung phone it should be some OneUI version. Also the latest update version/date or security patch level (if you can find this info).

Please ensure you have the latest updates installed.

**Comment by aniston** [comment]
Hallo @pastk im outside now, but here are my screenshots  of the Android.  Yes I have the latest security updates as an update  check results with no more updates. 

![Screenshot_20220611-204446_Settings](https://user-images.githubusercontent.com/3067891/173201034-96818059-d833-4792-9c15-c6854b515c94.jpg)
![Screenshot_20220611-204453_Settings](https://user-images.githubusercontent.com/3067891/173201037-989e7120-dae2-4a71-bda4-950c5df36e1f.jpg)

**Comment by pastk** [comment]
@aniston thanks!

We have one more idea to test (requesting run-time storage permissions), I hope to have a new beta ready tomorrow.

In the meantime could you please check on your phone if there are any SD-card related settings that could possibly affect its usage by apps? E.g. there is a Samsung-specific "Knox" security system, which might have some security settings/policies related to SD cards.

**Comment by pastk** [comment]
@aniston 

Here is beta5.
The difference is that when you enter "Save maps to" settings android will ask for storage access - grant it.
Then try to download a small map, restart the app, enter "Save maps to" again (it shall not ask second time) and send logs as usual.

Hope it helps!

**Comment by aniston** [comment]
Hi @pastk , this is the earliest I can report from beta5 !
- apart from giving permissions (allow) to storage access (granted) it behaves the same way as the last Betas 3 & 4

anyways log file is here [logsBeta5.zip](https://github.com/organicmaps/organicmaps/files/8899938/logsBeta5.zip)

my observations are as such::
- every time the OMb5 opens it asks to download the WorldMap, which is downloaded, but the App cannot read it, so background overlay and downloaded Maps list are empty.
- same with actually selecting and downloading a Map, Map gets downloaded (is listed and can be used in the App) but as soon as the App is closed and reopened it starts all over again ie: no WorldMap and no previously downloaded Maps, so basically back to square one.
- This behavior is because it cannot read the saved maps, I suggest we focus on the Map listing/reading part attributes/permissions specifically from SD card.
- and to confirm that the Maps are there , removing and reading the SD card on a PC shows us this, 
![image](https://user-images.githubusercontent.com/3067891/173585724-e4e1ddfc-928e-4f7f-bc09-de8efd0d73d0.png)

**Comment by pastk** [comment]
@aniston could you please check if there are any updates for you smartphone?

@bbinet says the latest update has fixed the issue for him! Bruno. what is the new security patch level you have now?

**Comment by bbinet** [comment]
![version](https://user-images.githubusercontent.com/76281/175503689-6a83d284-e735-473f-9b49-2ad889f9727e.png)

**Comment by pastk** [comment]
To re-cap / wrap up:
* it seems to be happening on some Samsung devices with Android 11 only
* happens both on stock and on LineageOS custom ROM
* the june update to LineageOS has fixed the problem
* there is hope that the next Samsung update will fix it for stock ROM too, but there is no guarantee as  it seems Samsung is releasing security updates only
* so a better approach would be to report this issue to Samsung directly

Technical details:
* OM uses the app-specific directory on SD card `Android/data/app.organicmaps/files/` returned by the android system call `Context.getExternalFilesDir(null)`
* according to [the docs](https://developer.android.com/reference/androidx/core/content/ContextCompat#getExternalFilesDirs(android.content.Context,%20java.lang.String)) all dirs returned by this call should be always accessible to the app (readable and writable without requesting any additional permissions)
* however in this case system calls `File.canRead()/canWrite()` return `false`
* but despite it the dir is both readable and writable - its possible to create and then read test directories and files, `File.exists()` works also
* but `File.list()/fileList()` return `null` (both for the main dir and subdirs), which means the app can't get list of dir's contents
* it works similarly for system calls from the native c++ code - files and directories are created and read, but listing doesn't work
* requesting additional run-time permissions (READ/WRITE_EXTERNAL_STORAGE) doesn't help

**Comment by pastk** [comment]
Hi @aniston,
if you have opportunity please test the following build N10.

https://drive.google.com/file/d/1Q4aip5SNmCRblOO4yUxrr_8YwAxLwh78/view?usp=sharing

We have one more idea to try to delete and re-create the dir returned by the system `Android/data/app.organicmaps.debug/files/` with a hope that it'll fix the access issue.

Please do a fresh install (uninstall a previous debug version first if you had it), then install, turn on logs, go to "Save maps to" menu and send the logs in, thanks!

Also a related idea you can try for yourself is to delete and re-create this dir manually (via a filemanager if you have access rights or via a computer perhaps).

**Comment by biodranik** [comment]
@bbinet @aniston did you try to do `ls -la` for Android/data/app.organicmaps.debug/files directory via `adb shell`? What were the permissions and owner?
 
Also, did you try to remount the SD-card?

**Comment by aniston** [comment]
> Hi @aniston, if you have opportunity please test the following build N10.
> 
> https://drive.google.com/file/d/1Q4aip5SNmCRblOO4yUxrr_8YwAxLwh78/view?usp=sharing
> 
> We have one more idea to try to delete and re-create the dir returned by the system `Android/data/app.organicmaps.debug/files/` with a hope that it'll fix the access issue.
> 
> Please do a fresh install (uninstall a previous debug version first if you had it), then install, turn on logs, go to "Save maps to" menu and send the logs in, thanks!
> 
> Also a related idea you can try for yourself is to delete and re-create this dir manually (via a filemanager if you have access rights or via a computer perhaps).

@pastk , regarding Debug V10, I uninstalled the last Beta v5 and verified that the app folder exists no more `sdcard/Android/data/app.organicmaps.debug` , after that installed the V10 called _Debug_

sadly this time round is the option to save to SD Card not available as you can see here.
![image](https://user-images.githubusercontent.com/3067891/177647441-e88ba2ce-3ce9-4ddd-81e8-6a62091cf72d.png)

- anyway attached is [logsV10.zip](https://github.com/organicmaps/organicmaps/files/9058122/logsV10.zip)

**Comment by pastk** [comment]
Thanks for the logs - the idea didn't work out, unfortunately.

As for the `ls` info:

`/sdcard` is misleading in Android, its internal memory actually (emulated storage), hence there are map files there - its working fine in your device.

Please add `ls -als` output for `/storage/54AE-180C/Android/data/app.organicmaps.debug` (its your SD card) and `files` dir there.

**Comment by aniston** [comment]
@biodranik , I noticed that the path starting with `/sdcard/` is misleading as that is not the real SD card path !
- instead `/storage/54AE-180C/` is the correct SD card path and effectively is empty :( ie: no files are getting stored there for `Android/data/app.organicmaps.debug`
- maybe the following is of more interest to you:
```
xcover4s:/ $ ls -als /storage/54AE-180C/Android/data/app.organicmaps.debug
total 32
32 drwxrwx--- 2 u0_a326 everybody 32768 2022-07-06 23:33 files

xcover4s:/ $ ls -als /storage/54AE-180C/Android/data/app.organicmaps.debug/files/
total 0
```

**Comment by aniston** [comment]
@pastk , just wondering how you manage to read so quickly the GitHub Issue messages ?
- is there a better way than the standard message list direct on GitHub ?

**Comment by Bradnik** [comment]
I'm having the same problem with a Moto e30 using Android 11. Until this is fixed, Organic Maps is unusable for me, and I shall have to go back to Maps.me )-:

**Comment by biodranik** [comment]
@Bradnik Am I correctly understanding that MapsMe properly saves maps to your device's SD card? What is your device model and Android version?

**Comment by Bradnik** [comment]
Actually it doesn't. It seemed to save OK (didn't say read-only), but a few days later the maps had disappeared.

On 24 September 2022 23:00:05 BST, Alexander Borsuk ***@***.***> wrote:
***@***.*** Am I correctly understanding that MapsMe properly saves maps to your device's SD card?

**Comment by biodranik** [comment]
@Bradnik You still have the same issue with Organic Maps (OM), right? Can you please enable logs in OM settings, restart OM, open settings/SD card dialog there, and then share logs with us via "?" -> Report a bug?

**Comment by biodranik** [comment]
@SK53 that's interesting. But I assume that your other device won't have any issues with the latest Organic Maps version. From our users we see that it starts after some system firmware upgrade.

It would be great if you back up your bookmarks (export them), update OM on that other device, and tell us if it works. If it doesn't work (highly unlikely), then you can always install again an older version from FDroid. This experiment is very important to prove or disprove that there is an issue in our current code. Thank you in advance!

**Comment by SK53** [comment]
@biodranik OK, I uninstalled the Fdroid version & reinstalled from Playstore 2023.02.13-11-Google. Although uninstalled it used existing settings and downloaded maps to the SD Card! The tablet has an Android kernel version from January, the phone from August. I'm now going to try & update the basic software on the phone.

**Comment by biodranik** [comment]
So you confirm that the latest OM version works correctly with the SD card on another device, correct?

**Comment by SK53** [comment]
@biodranik Yes, correct. Further testing shows odd behaviour of other apps with the SD Card, so it does appear to be related to my device (and is something which is quite recent). For instance I cannot move Vespucci to run on the SD Card: something I have done in the past.

**Comment by matkoniecz** [comment]
> [Won't fix](https://github.com/organicmaps/organicmaps/labels/Won%27t%20fix)

then maybe it can be closed?

(or if it is wontfix then label should be probably removed)
